### PR TITLE
[WIP] Add PHAR build support with Box: https://github.com/box-project/box2

### PR DESCRIPTION
**Pull request by padraic**
Box installation instructions in its README. Then...

```
mkdir build
./path/to/box(.phar) build
# test...
./build/phpmnd.phar ./src
```
Should be easy enough to add a self-update command tracking PHAR files attached to future Github releases. I'll try taking a look at that in a while.

Implements #15.

### CODE REVIEW

File: box.json.dist [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                "theseer/tokenizer",
+                "sebastian/object-reflector"
+            ],
+            "in": [
+                "vendor"
+            ]
+        }
+    ],
+    "git-version": "package_version",
+    "main": "phpmnd",
```
Comment (padraic) [inline box.json.dist]: Change to `bin/phpmnd` if #34 merged.

File: box.json.dist [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                "phpunit/php-text-template",
+                "doctrine/instantiator",
+                "sebastian/diff",
+                "sebastian/resource-operations",
+                "sebastian/version",
+                "phar-io/manifest",
+                "phar-io/version",
+                "theseer/tokenizer",
+                "sebastian/object-reflector"
+            ],
```
Comment (padraic) [inline box.json.dist]: List compiled by `composer update` and then reading off the removed packages running `composer update --no-dev` which presumably tells you exactly what is NOT needed to be included in distributable PHARs.

File: .gitattributes [hunk 3]
```diff
@@ -7,3 +7,5 @@ LICENSE export-ignore
 phpunit.xml.dist export-ignore
 README.md export-ignore
 tests/ export-ignore
+box.json.dist export-ignore
+build/ export-ignore
```
Comment (povils) [inline .gitattributes]: newlines

**Comment by povils** [comment]
thanks!

**Comment by padraic** [comment]
No problem! I'll look into self-updates in a few days.

**Pull request merged**